qt/scientific_interfaces/MultiDatasetFit/MultiDatasetFit.cpp

```diff
@@ -25,6 +28,49 @@ namespace {
 const int zoomToolPage = 0;
 const int rangeToolPage = 1;
 Mantid::Kernel::Logger g_log("MultiDatasetFit");
+
+/// Copy parameter values into a table workspace that is ready for
+/// plotting them against a dataset index.
+void formatParametersForPlotting(const Mantid::API::IFunction &function,
+                                 const std::string &parametersPropertyName) {
+  const auto &mdFunction =
+      dynamic_cast<const Mantid::API::MultiDomainFunction &>(function);
+
+  if (mdFunction.nFunctions() == 0) {
+    // Fit button was hit by mistake? Nothing to do here.
+    // A warning will be shown elsewhere.
+    return;
+  }
+
+  auto table =
+      Mantid::API::WorkspaceFactory::Instance().createTable("TableWorkspace");
+  auto col = table->addColumn("double", "Dataset");
+  col->setPlotType(1); // X-values inplots
+
+  auto nDomains = mdFunction.getNumberDomains();
+  assert(nDomains == mdFunction.nFunctions());
+
+  // Add columns for parameters and their errors
+  const auto &fun = *mdFunction.getFunction(0);
+  for (size_t iPar = 0; iPar < fun.nParams(); ++iPar) {
+    table->addColumn("double", fun.parameterName(iPar));
+    table->addColumn("double", fun.parameterName(iPar) + "_Err");
+  }
+
+  // Fill in the columns
+  for (size_t iData = 0; iData < nDomains; ++iData) {
+    Mantid::API::TableRow row = table->appendRow();
```

Comment:
Since the number of rows is known beforehand, why not create enough rows before this loop instead of appending at each iteration? `table->setRowCount(nDomains)` should do the job. This might be beneficial performance-wise.

Reply:
I don't think it's a big issue but I can do it.